File: ENTIDADES/Almacen/AInventario.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace ENTIDADES.Almacen
{
    [Table("Inventario", Schema = "Almacen")]
    public class AInventario : AuditoriaColumna
    {
        [Key]
        public int idinventario { get; set; }
        public int suc_codigo { get; set; }
        public int idalmacensucursal { get; set; }
        public int idlaboratorio { get; set; }
        public int usuarioinicia { get; set; }
        public DateTime fechainicio { get; set; }
        public int usuariofinaliza { get; set; }
        public DateTime fechafin { get; set; }
        public string estado { get; set; }

        [NotMapped]
        public string jsondetalle { get; set; }
    }
}

File: ERP/wwwroot/js/modelos/almacen/AInventario.js
class AInventario {
    constructor() {
        this.idinventario = 0;
        this.suc_codigo = 0;
        this.idalmacensucursal = 0;
        this.usuarioinicia = 0;
        this.fechainicio = "";
        this.usuariofinaliza = 0;
        this.fechafin = "";
        this.estado = "";
    }
}

class AInventarioDetalle {
    constructor() {
        this.idinventariodetalle = 0;
        this.idinventario = 0;
        this.idstock = 0;
        this.cajas = 0;
        this.unidad = 0;
        this.estado = "";
    }
}

class AStockLoteProducto {
    constructor() {
        this.idstock = 0;
        this.idproducto = 0;
        this.idalmacensucursal = 0;
        this.lote = "";
        this.regsanitario = "";
        this.fechavencimiento = "";
        this.idtabla = 0;
    }
}